samples/examples/FSharpExamples/Matrix.fs

```diff
@@ -28,38 +28,53 @@ open Prajna.Api.FSharp
 
 open Prajna.Examples.Common
 
-type MatrixSample() =
+module Matrix =
 
     let (.*) (xs: float[]) (ys: float[]) = 
-        // simulate doing some more work, 
-        // e.g.: run the remaining layers of a neural net
-        Threading.Thread.Sleep(3) 
         Array.fold2 (fun sum x y -> sum + x * y) 0.0 xs ys 
 
     let printTiming (msg: string) (x: Lazy<'b>) : 'b =
         let sw = Stopwatch.StartNew()
+        printf "%s: " msg
         let ret = x.Value
         sw.Stop()
-        printfn "%s: %A" msg sw.Elapsed
+        printfn "%A" sw.Elapsed
         ret
 
-    let run (cluster: Cluster) =
+    let random = new Random(0)
+
+    let randomVector n =  Array.init n (fun _ -> random.NextDouble())
+    
+    let randomMatrix m n =  Array.init m (fun _ -> randomVector n)
+
+    /// Matrix-Vector multiplication for illustrational purposes.
+    /// The overhead of distributing makes it faster to run this locally.
+    /// See Matrix-Matrix for a case where the computation is actually faster distributed.
+    let matrixVector (cluster: Cluster) =
 
         let numRows = 200
         let vectorSize = 100
 
-        let r = new Random(0)
-        let randomMatrix =  Array.init numRows (fun _ -> Array.init vectorSize (fun _ -> r.NextDouble()))
-        let vector = Array.init vectorSize (fun _ -> r.NextDouble())
+        let matrix =  randomMatrix numRows vectorSize
+        let vector = randomVector vectorSize
 
         let singleCoreResult = 
             printTiming "single core" <| lazy 
-                randomMatrix |> Array.map (fun row -> row .* vector)
-        
+                matrix |> Array.map (fun row -> row .* vector)
+
+        let numPartitions = Environment.ProcessorCount
+        let dSet =
+            printTiming "distributing" <| lazy
+                let d =
+                    DSet<float[]>(Cluster = cluster, NumParallelExecution = numPartitions)
+                    |> DSet.distributeN numPartitions matrix
```

Comment:
I tried, but it doesn't have the same effect, at least not for now, and it was seeming to things worse, and using fold does what I wanted. To be sure, after your comment (maybe I was imagining things), I ran an experiment: First, to set a baseline, I ran 5 times without calling cacheInMemory or fold. Here are the results:

single core: 4.107
distributing: 0.004
computing distributed: 1.407

single core: 4.806
distributing: 0.004
computing distributed: 1.328

single core: 4.556
distributing: 0.004
computing distributed: 1.325

single core: 4.667
distributing: 0.004
computing distributed: 1.351

single core: 3.924
distributing: 0.004
computing distributed: 1.356

Next I used cacheInMemory. Eventually, we should make this work well, perhaps under a different name. But for now, here's what I get when I add a "|> DSet.cacheInMemory":

single core: 4.711
distributing: 0.028
computing distributed: 1.487

single core: 4.568
distributing: 0.012
computing distributed: 1.465

single core: 4.62
distributing: 0.013
computing distributed: 1.382

single core: 4.096
distributing: 0.013
computing distributed: 1.399

single core: 4.075
distributing: 0.013
computing distributed: 1.425

Finally, to make sure the fold was actually doing something, I ran 5 times using fold, as in the current code. Here's what I get:

single core: 4.03
distributing: 0.634
computing distributed: 1.061

single core: 4.476
distributing: 0.63
computing distributed: 1.162

single core: 4.158
distributing: 0.625
computing distributed: 1.138

single core: 4.783
distributing: 0.706
computing distributed: 1.087

single core: 4.787
distributing: 0.588
computing distributed: 1.145

So adding a cacheInMemory instead of fold makes things ~30%-50% slower when running distributed. We should fix this, of course, but the purpose of the sample is to demonstrate how you can use Prajna to do a distributed Matrix-Matrix multiply with minimum overhead, and the straightforward way to do this for now is adding a dummy fold before the main computation.